src/EnergyPlus/InputProcessing/InputProcessor.cc

```diff
@@ -542,6 +542,82 @@ bool InputProcessor::getDefaultValue(EnergyPlusData &state, std::string const &o
     return defaultFound;
 }
 
+std::string InputProcessor::getAlphaFieldValue(
+    EnergyPlusData &state, std::string const &objectWord, json const &ep_object, json const &schema_obj_props, std::string const &fieldName)
+{
+    auto const &schema_field_obj = schema_obj_props[fieldName];
+    if (schema_field_obj.empty()) {
+        ShowFatalError(state, "InputProcessor::fieldValue: Invalid field name = " + fieldName + "for object type = " + objectWord);
+    }
+    bool isDefaulted = false;
+    std::string value;
+    auto it = ep_object.find(fieldName);
+    if (it != ep_object.end()) {
+        auto const &field_value = it.value();
+        if (field_value.is_string()) {
+            auto valuePair = getObjectItemValue(field_value.get<std::string>(), schema_field_obj);
+            value = valuePair.first;
+            isDefaulted = valuePair.second;
+        } else {
+            ShowSevereError(state, "InputProcessor::fieldValue: Invalid field type = " + fieldName + "for object type = " + objectWord);
+            ShowFatalError(state, "String value requested but field type is numeric");
+        }
+    } else {
+        isDefaulted = findDefault(value, schema_field_obj);
+        if (!isDefaulted) {
+            value = "";
+        }
+    }
+    return value;
+}
+
+Real64 InputProcessor::getRealFieldValue(
+    EnergyPlusData &state, std::string const &objectWord, json const &ep_object, json const &schema_obj_props, std::string const &fieldName)
```

Comment:
Does it make sense to have separate functions for `Real` and `Int`?

Reply:
Good question. I started to do that and then realized that the epJSON schema does not differentiate between real and integer: "type": "number"

Reply:
getRealFieldVal could be changed to getNumericFieldVal if there is no differentiation between int and real. I wonder if returning a real will have any effect when reading into an int variable but since we do that now anyway with getObjectItem's Numbers(x) it shouldn't be a problem.

Reply:
If you wanted to return `int` or `Real`, you would need to have separate functions since they'd likely have the same input parameter signature and only change the return type.

Reply:
I will also say with the following, which I saw in an outdated comment in this PR:
```
Real64 coeff = 0.0;
state.dataInputProcessing->inputProcessor->getFieldValue(state, CurrentModuleObject, fields, objectSchemaProps, "air_mass_flow_coefficient_at_reference_conditions", coeff);
```
It does seem odd to not directly return the fieldValue, however, this function could be overloaded to include `int` or `string` as input types with their own unique functions. Such as below:
```
int coeff = 0.0;
state.dataInputProcessing->inputProcessor->getFieldValue(state, CurrentModuleObject, fields, objectSchemaProps, "air_mass_flow_coefficient_at_reference_conditions", coeff);

std::string coeff = "look a coefficient!";
state.dataInputProcessing->inputProcessor->getFieldValue(state, CurrentModuleObject, fields, objectSchemaProps, "air_mass_flow_coefficient_at_reference_conditions", coeff);
```
This means there is only ever one function name to call.
Otherwise, you'll need to have `getRealFieldValue`, `getIntFieldValue`, and `getAlphaFieldValue` functions like below:
```
Real64 coeff = state.dataInputProcessing->inputProcessor->getRealFieldValue(state, CurrentModuleObject, fields, objectSchemaProps, "air_mass_flow_coefficient_at_reference_conditions");
```